grails-app/domain/au/org/emii/aatams/ReceiverDeployment.groovy

```diff
@@ -140,46 +133,99 @@ class ReceiverDeployment
         return true
     }
 
-    String toString()
-    {
-        return String.valueOf(receiver) + " - " + String.valueOf(deploymentDateTime)
+    static def conflictingDeploymentValidator = { initialisationDateTime, deployment ->
+        // All non-null deployment intervals for receiver, other than 'this' deployment's.
+        def conflictingDeployments = deployment.receiver?.deployments.findAll {
+            !it.same(deployment) && it.undeployableInterval
+        }.findAll {
+            def checkInterval = it.undeployableInterval
+            checkInterval.contains([
+                deployment.initialisationDateTime,
+                deployment.deploymentDateTime
+            ]) ||
+
+            // deployment.deployedInterval?.overlaps(checkInterval)
+            deployment.undeployableInterval?.overlaps(checkInterval)
```

Comment:
I think this return is a bit deceptive. It might be easy to miss the `||` from above. Maybe stick both parts in variables and then finish with ORing them?

Also, I think this comment can go.